sdk/Pulumi/Resources/CustomTimeouts.cs

```diff
@@ -31,5 +31,128 @@ public sealed class CustomTimeouts
                 Delete = timeouts.Delete,
                 Update = timeouts.Update,
             };
+
+
+        private static string TimeoutString(TimeSpan? timeSpan)
+        {
+            if (timeSpan == null)
+                return "";
+
+            // This will eventually be parsed by go's ParseDuration function here:
+            // https://github.com/pulumi/pulumi/blob/06d4dde8898b2a0de2c3c7ff8e45f97495b89d82/pkg/resource/deploy/source_eval.go#L967
+            //
+            // So we generate a legal duration as allowed by
+            // https://golang.org/pkg/time/#ParseDuration.
+            //
+            // Simply put, we simply convert our ticks to the integral number of nanoseconds
+            // corresponding to it.  Since each tick is 100ns, this can trivially be done just by
+            // appending "00" to it.
+            return timeSpan.Value.Ticks + "00ns";
+        }
+
+        internal Pulumirpc.RegisterResourceRequest.Types.CustomTimeouts Serialize()
+        {
+            return new Pulumirpc.RegisterResourceRequest.Types.CustomTimeouts
+            {
+                Create = TimeoutString(Create),
+                Update = TimeoutString(Update),
+                Delete = TimeoutString(Delete),
+            };
+        }
+
+        internal static CustomTimeouts Deserialize(Pulumirpc.RegisterResourceRequest.Types.CustomTimeouts customTimeouts)
+        {
+            Func<string, TimeSpan?> parse = (string s) =>
```

Comment:
Nit: This could be a [local function](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/local-functions).